### PR TITLE
Pro 3213 add fileGroups documentation

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR adds documentation for the new `addFileGroups` option of the `attachment` module. Closes PRO-3213.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Build the documentation.
2. Navigate to '/reference/modules/attachment.html' and '/reference/field-types/attachment.html'

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
Will not be deployed until the release of the `attachment` module update.

### CODE REVIEW

File: docs/reference/field-types/attachment.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ The uploaded files are stored in a web-accessible folder, however their file nam
 
 ## Custom file groups
 
-Developers can configure file type groups in addition to `office` and `image` using the `fileGroups` option of the `@apostrophecms/attachment` module. Those custom groups names can then be used for an attachment field's `fileGroup` setting.
+Developers can configure file type groups in addition to `office` and `image` using the `fileGroups` or `addFileGroups` options of the [`@apostrophecms/attachment` module](./../modules/attachment.md). Those custom groups names can then be used for an attachment field's `fileGroup` setting.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/field-types/attachment.md:52]: Changed.

File: docs/reference/modules/attachment.md [hunk 2]
```diff
@@ -77,7 +79,80 @@ The first default object's `name` property is set to `images` and an `extensions
...
 :::
 
-The second default object is very similar, but for the `name` key it takes a value of `office`, and `image: false`, indicating that these file types can be attached to `@apostrophecms/file` pieces and should not be processed. Again, the `extensions` key takes an array of non-prefixed strings indicating the allowed file types. The `extensionMaps` maps alternative spellings to the specified extension.
+The second default object is very similar, but for the `name` key it takes a value of `office`, indicating that these file types can be attached to `@apostrophecms/file` pieces and should not be processed. Again, the `extensions` key takes an array of non-prefixed strings indicating the allowed file types. The `extensionMaps` maps alternative spellings to the specified extension.
+
+### `addFileGroups`
+
+This option should be passed to the `@apostrophecms/attachment` moduleand it takes an array of objects. Unlike the `fileGroups` option, this option is used to add to, not replace, the existing default extension groups.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/modules/attachment.md:86]: Changed.

File: docs/reference/modules/attachment.md [hunk 3]
```diff
@@ -77,7 +79,80 @@ The first default object's `name` property is set to `images` and an `extensions
...
+In addition to `name`, the object can have both `extensions` and `extensionMaps` properties that take values just like the properties in the `fileGroups` options. Both are optional, you only are required to supply one, but can supply both.
+
+The `addFileGroups` option can also be used to add a new grouping without eliminating the existing groups. This is accomplished through the addition of a new object containing `name`, `label`, `extensions` and `extensionMaps` properties. All are required.
+
+### Example
+
+<AposCodeBlock>
+
+```javascript
+module.exports= {
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/modules/attachment.md:128]: Changed.

File: docs/reference/modules/attachment.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/modules/attachment.md]: Changed

File: docs/reference/modules/attachment.md [hunk 4]
```diff
@@ -238,6 +238,12 @@ The `options` parameter is optional. If `options.permissions` is explicitly set
 ### `crop(req, _id, crop)`
 This method takes the image specified by the `_id` of an existing attachment, copies it to the uploadfs specified temporary location, applies the crop, and then saves it back into uploadfs storage. The `crop` parameter takes an object with `top`, `left`, `width`, and `height` properties. The passed values should be unitless, but must be JavaScript numbers, not strings.
 
+### `addFieldGroup(group)`
```
Comment (boutell) [inline docs/reference/modules/attachment.md:241]: `addFileGroup` right?

File: docs/reference/modules/attachment.md [hunk 5]
```diff
@@ -238,6 +238,12 @@ The `options` parameter is optional. If `options.permissions` is explicitly set
...
+This method takes an object specifying either a new file group or that extends an existing group. For new file groups the object should have `name`, `extensions`, and `extensionMaps` properties that are set-up like the objects passed through array to the `fieldGroups` option of this module.
+
+To extend an existing file group the object should have a `name` property matching the `name` value of an existing file group (`office` or `images`, by default). The object should also have an `extensions` property, `extensionMaps` property, or both. These properties are structured like the properties of the same name in the `fieldGroups` objects.
```
Comment (boutell) [inline docs/reference/modules/attachment.md:244]: `fileGroups` right?

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/reference/modules/attachment.md (hunk 4 above)
Comment (BoDonkey) [reply docs/reference/modules/attachment.md:241]: Dammit! My "muscle memory" keeps defaulting to that.

**Review by boutell — APPROVED** [review]

**Pull request merged**